### PR TITLE
Add regression test

### PR DESCRIPTION
**Pull request by Frost45**
Add script for running regression tests.

Once this PR is merged, we are ready to submit our first PR to the upstream repo (Grammarly's GECToR repo).

To test, run `python regression_tests/test_gector_roberta.py` from the base `gector` directory.

### CODE REVIEW

File: regression_tests/test_gector_roberta.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    """
+    Downloads model weights from S3 if not already present at path.
+
+    Returns
+    -------
+    Path
+        Path to model weights file
+    """
+
+    model_url = "https://grammarly-nlp-data-public.s3.amazonaws.com/gector/roberta_1_gectorv2.th"
```
Comment (desilinguist) [inline regression_tests/test_gector_roberta.py:25]: This can be a constant, right?

File: regression_tests/test_gector_roberta.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            preds, cnt = model.handle_batch(batch)
+            predictions.extend(preds)
+            cnt_corrections += cnt
+            batch = []
+    if batch:
+        preds, cnt = model.handle_batch(batch)
+        predictions.extend(preds)
+        cnt_corrections += cnt
+
+    result_lines = [" ".join(x) for x in predictions]
```
Comment (desilinguist) [inline regression_tests/test_gector_roberta.py:76]:
```suggestion
    result_lines = [" ".join(pred) for pred in predictions]
```

File: regression_tests/test_gector_roberta.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    assert filecmp.cmp(
+        gold_file, temp_file, shallow=False
+    ), f"Output of {filename} does not match gold output."
+    print(filename, "passed.")
+
+
+def predict_and_compare(model):
+    """
+    Generate predictions for all test files and tests that there are no changes.
```
Comment (desilinguist) [inline regression_tests/test_gector_roberta.py:106]:
```suggestion
    Generate predictions for all test files and test that there are no changes.
```

File: regression_tests/test_gector_roberta.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    Parameters
+    ----------
+    model : GecBERTModel
+        Initialized model
+    """
+
+    for child in ORIG_FILE_DIR.iterdir():
+        if child.is_file():
+            input_file = str(ORIG_FILE_DIR.joinpath(child.name))
```
Comment (desilinguist) [inline regression_tests/test_gector_roberta.py:116]:
```suggestion
            input_file = str(child.resolve())
```

**Review by desilinguist — CHANGES_REQUESTED** [review]
Looks great! Just a few minor comments.

**Review by desilinguist — APPROVED** [review]
LGTM!

File: regression_tests/test_gector_roberta.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        preds, cnt = model.handle_batch(batch)
+        predictions.extend(preds)
+        cnt_corrections += cnt
+
+    result_lines = [" ".join(pred) for pred in predictions]
+
+    with open(temp_file.name, "w") as f:
+        f.write("\n".join(result_lines) + "\n")
+
+    return cnt_corrections
```
Comment (damien2012eng) [inline regression_tests/test_gector_roberta.py:81]: The `cnt_corrections` is not used anywhere. Should we just delete this line?

**Comment by damien2012eng** [comment]
Looks great to me! Just have a nitpick.

**Comment by damien2012eng** [comment]
One more favor to ask: in the `test_tokenization.py`, the class name is incorrect. The class name should start with `Test` (detailed discussion [here](https://stackoverflow.com/questions/37353960/pytest-exits-with-no-error-but-with-collected-0-items)), otherwise pytest will not able to gather tests. So could you please change the class name from `TokenizationTests` to `TestTokenization`?

**Review by damien2012eng — APPROVED** [review]

**Pull request merged**